update VolumeGroupSnapshot to v1beta1

Madhu-1 opened the pull request:
Updating the required files to move VGS from alpha to beta API.

Madhu-1 committed:
build: update snapshot version

updating the snapshot version in
the build.env

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: update template for vgs flag

updating the snapshotter image to 8.2.0
and update the flag to enable vgs
feature.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
helm: update template for vgs flag

updating the snapshotter image to 8.2.0
and update the flag to enable vgs
feature.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
e2e: update snapshot to v1beta1

updating the snapshot client and
test to use v1beta1 client and API.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: adjust the snapshot to enable vgs

updating the snapshot container ARG
to enable the vgs feature.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: update the vgs api to v1beta1

updating the yamls of vgs to v1beta1

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/mini-e2e/k8s-1.31

nixpanic commented on scripts/install-snapshot.sh:
```diff
@@ -78,12 +78,12 @@ function create_or_delete_resource() {
     if [ "${operation}" == "create" ]; then
         # Argument to add/update
-        ARGUMENT="--enable-volume-group-snapshots=true"
+        ARGUMENT="--feature-gates=CSIVolumeGroupSnapshot=true"
```
this commit message mentions _vsg_ instead of **vgs**, please correct that

nixpanic reviewed (commented):
Just a nit, all other things look good to me.

nixpanic commented:
@mergifyio queue

nixpanic commented:
Fixed a few remaining v8.0.1 references (NFS deployment and helm values.yaml).

iPraveenParihar commented:
@Madhu-1 

upgrade tests are failing - 
```
  I1213 12:14:48.098122       1 reflector.go:332] Listing and watching *v1alpha1.VolumeGroupSnapshotContent from github.com/kubernetes-csi/external-snapshotter/client/v8/informers/externalversions/factory.go:142
  W1213 12:14:48.099323       1 reflector.go:547] github.com/kubernetes-csi/external-snapshotter/client/v8/informers/externalversions/factory.go:142: failed to list *v1alpha1.VolumeGroupSnapshotContent: the server could not find the requested resource (get volumegroupsnapshotcontents.groupsnapshot.storage.k8s.io)
  E1213 12:14:48.099361       1 reflector.go:150] github.com/kubernetes-csi/external-snapshotter/client/v8/informers/externalversions/factory.go:142: Failed to watch *v1alpha1.VolumeGroupSnapshotContent: failed to list *v1alpha1.VolumeGroupSnapshotContent: the server could not find the requested resource (get volumegroupsnapshotcontents.groupsnapshot.storage.k8s.io)
  ```
  
  install-snapshot.sh installs beta CRDs and upgrade test using 3.12.1 version deploys older snapshotter sidecar.

Madhu-1 committed:
e2e: remove vsg flag present in older snapshotter

Remove the vsg flag present in the older snapshotter
which is meant only for the Alpha Release and we are
no longer installing Alpha CRD.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/upgrade-tests-rbd

Madhu-1 commented:
/test ci/centos/mini-e2e/k8s-1.30

Madhu-1 commented:
> @Madhu-1
> 
> upgrade tests are failing -
> 
> ```
>   I1213 12:14:48.098122       1 reflector.go:332] Listing and watching *v1alpha1.VolumeGroupSnapshotContent from github.com/kubernetes-csi/external-snapshotter/client/v8/informers/externalversions/factory.go:142
>   W1213 12:14:48.099323       1 reflector.go:547] github.com/kubernetes-csi/external-snapshotter/client/v8/informers/externalversions/factory.go:142: failed to list *v1alpha1.VolumeGroupSnapshotContent: the server could not find the requested resource (get volumegroupsnapshotcontents.groupsnapshot.storage.k8s.io)
>   E1213 12:14:48.099361       1 reflector.go:150] github.com/kubernetes-csi/external-snapshotter/client/v8/informers/externalversions/factory.go:142: Failed to watch *v1alpha1.VolumeGroupSnapshotContent: failed to list *v1alpha1.VolumeGroupSnapshotContent: the server could not find the requested resource (get volumegroupsnapshotcontents.groupsnapshot.storage.k8s.io)
> ```
> 
> install-snapshot.sh installs beta CRDs and upgrade test using 3.12.1 version deploys older snapshotter sidecar.

fixed now, it's a weird situation. we will remove it in next release

nixpanic commented:
@mergifyio queue
